### PR TITLE
Brian/function-calling

### PR DESCRIPTION
**Pull request by jerhadf**
merge in function calling example

### CODE REVIEW

File: evi-next-js-function-calling/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      "version": 0
+    }
+  ]
+}
+```
+
+3. Add the Config ID to your environment variables.
+
+```bash
+echo "NEXT_PUBLIC_HUME_CONFIG_ID= <YOUR HUME CONFIG ID>" >> .env
```
[the remark this answers is not in the document]
Comment (jerhadf) [reply evi-next-js-function-calling/README.md:47]: fixed!

File: evi-next-js-function-calling/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "evi-next-js-app-router",
```
Comment (zachkrall) [inline evi-next-js-function-calling/package.json:2]:
```suggestion
  "name": "evi-next-js-function-calling",
```

File: evi-next-js-function-calling/src/components/ClientComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        properties: {
+          forecast: string;
+        };
+      };
+      const { properties } = json;
+      const { forecast: forecastUrl } = properties;
+
+      const forecastResult = await fetch(forecastUrl);
+
+      const forecastJson = (await forecastResult.json()) as {
```
Comment (zachkrall) [inline evi-next-js-function-calling/src/components/ClientComponent.tsx:45]: i think @yinishi made some changes here in the other repo that we should carry over
https://github.com/HumeAI/empathic-voice-api-js/pull/189

File: evi-next-js-function-calling/src/components/ClientComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        tool_call_id: toolCall.tool_call_id,
+        content: JSON.stringify(forecast),
+      };
+    } catch (error) {
+      return {
+        type: 'tool_error',
+        tool_call_id: toolCall.tool_call_id,
+        error: 'Weather tool error',
+        code: 'weather_tool_error',
+        level: 'error',
```
Comment (zachkrall) [inline evi-next-js-function-calling/src/components/ClientComponent.tsx:63]:
```suggestion
        level: 'warn',
```

File: evi-next-js-function-calling/src/components/ClientComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        content: 'There was an error with the weather tool',
+      };
+    }
+  } else {
+    return {
+      type: 'tool_error',
+      tool_call_id: toolCall.tool_call_id,
+      error: 'Tool not found',
+      code: 'tool_not_found',
+      level: 'warning',
```
Comment (zachkrall) [inline evi-next-js-function-calling/src/components/ClientComponent.tsx:73]:
```suggestion
      level: 'warn',
```

File: evi-next-js-function-calling/src/components/ClientComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
+"use client";
+import { VoiceProvider, ToolCall, ToolResponse, ToolError } from "@humeai/voice-react";
+import Messages from "./Controls";
+import Controls from "./Messages";
+
+const handleToolCall = async (
```
Comment (zachkrall) [inline evi-next-js-function-calling/src/components/ClientComponent.tsx:6]: the sdk provides a `ToolCallHandler` type that applies correct types to parameters and return values

File: evi-typescript-example/src/main.ts [hunk 7]
```diff
@@ -99,15 +98,16 @@ function getElementById<T extends HTMLElement>(id: string): T | null {
    * instantiates interface config and client, sets up Web Socket handlers, and establishes secure Web Socket connection
    */
   async function connect(): Promise<void> {
-    // creates minimal EVI configuration
-    const config = createConfig({
```
Comment (zachkrall) [inline evi-typescript-example/src/main.ts:103]: we should keep this utility in

File: evi-next-js-function-calling/src/components/ClientComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+      const forecastResult = await fetch(forecastUrl);
+
+      const forecastJson = (await forecastResult.json()) as {
+        properties: {
+          periods: unknown[];
+        };
+      };
+      const forecast = forecastJson.properties.periods;
+
+      return {
```
Comment (zachkrall) [inline evi-next-js-function-calling/src/components/ClientComponent.tsx:52]: using the `response.success()` and `response.error()` helpers here would be useful for documentation https://github.com/HumeAI/empathic-voice-api-js/pull/191

File: evi-next-js-function-calling/package.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@humeai/voice": "0.1.3-beta.1",
+    "@humeai/voice-react": "0.1.3-beta.1",
```
Comment (zgreathouse) [inline evi-next-js-function-calling/package.json:13]: @hume-brian Can you confirm whether these package versions are publicly available? (private vs public NPM registry) 

We'll want to make sure the packages are installable by external devs attempting to run the project locally.

File: evi-typescript-example/package.json [hunk 10]
```diff
@@ -9,7 +9,7 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@humeai/voice": "0.0.0-beta.20"
+    "@humeai/voice": "^0.1.2"
```
Comment (zgreathouse) [inline evi-typescript-example/package.json:12]: @jerhadf We can remove the `evi-typescript-example` project changes from this branch. Some of these changes are already in. I am currently working on it in another branch to utilize the Fern typescript SDK.

File: evi-next-js-function-calling/package.json (hunk 9 above)
Comment (yinishi) [reply evi-next-js-function-calling/package.json:13]: all packages are now on 0.1.3 as of this morning

File: evi-typescript-example/package.json (hunk 10 above)
Comment (jerhadf) [reply evi-typescript-example/package.json:12]: okay sg could you remove those changes please? 🙏 sorry working on some other pre-release tool use things

File: evi-next-js-function-calling/.env.example [hunk 11]
```diff
@@ -0,0 +1,4 @@
+HUME_API_KEY=""
+HUME_CLIENT_SECRET=""
```
Comment (zgreathouse) [inline evi-next-js-function-calling/.env.example:2]: Unless I am mistaken, I think we should also prefix these env variables with `NEXT_PUBLIC_`

https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#bundling-environment-variables-for-the-browser

File: evi-typescript-example/package.json (hunk 10 above)
Comment (zgreathouse) [reply evi-typescript-example/package.json:12]: For sure!

**Pull request merged**